<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RpcTestDialog</class>
 <widget class="QWidget" name="RpcTestDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>680</width>
    <height>502</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>RpcTestDialog</string>
  </property>
  <widget class="QPushButton" name="pushButton">
   <property name="geometry">
    <rect>
     <x>100</x>
     <y>70</y>
     <width>121</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>RPC Server Start</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_2">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>70</y>
     <width>131</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>RPC Client Start</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_3">
   <property name="geometry">
    <rect>
     <x>100</x>
     <y>110</y>
     <width>121</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>RPC Server Stop</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_4">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>120</y>
     <width>141</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>RPC Client Stop</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_5">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>180</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>publish</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="textEdit">
   <property name="geometry">
    <rect>
     <x>60</x>
     <y>280</y>
     <width>201</width>
     <height>141</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>60</x>
     <y>260</y>
     <width>101</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Received Request:</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="textEdit_2">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>170</y>
     <width>141</width>
     <height>51</height>
    </rect>
   </property>
  </widget>
  <widget class="QTextEdit" name="textEdit_3">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>170</y>
     <width>131</width>
     <height>51</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_6">
   <property name="geometry">
    <rect>
     <x>540</x>
     <y>180</y>
     <width>101</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>RequestSend</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>250</y>
     <width>201</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Received reply and publish msg:</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="textEdit_4">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>280</y>
     <width>201</width>
     <height>141</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>20</y>
     <width>271</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>request-reply and publish/subscribe mode</string>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>RpcTestDialog</receiver>
   <slot>rpcserver_stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>202</x>
     <y>125</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>10</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>RpcTestDialog</receiver>
   <slot>rpcserver_start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>175</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>323</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>RpcTestDialog</receiver>
   <slot>rpcclient_start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>422</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>RpcTestDialog</receiver>
   <slot>rpcclient_stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>483</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>623</x>
     <y>29</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>RpcTestDialog</receiver>
   <slot>rpcserver_publish()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>280</x>
     <y>186</y>
    </hint>
    <hint type="destinationlabel">
     <x>362</x>
     <y>82</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>RpcTestDialog</receiver>
   <slot>rpcclient_sendreq()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>604</x>
     <y>186</y>
    </hint>
    <hint type="destinationlabel">
     <x>607</x>
     <y>125</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>rpcserver_start()</slot>
  <slot>rpcserver_stop()</slot>
  <slot>rpcclient_start()</slot>
  <slot>rpcclient_stop()</slot>
  <slot>rpcserver_publish()</slot>
  <slot>rpcclient_sendreq()</slot>
 </slots>
</ui>
